core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -187,7 +187,7 @@ class TransactionCoordinator(brokerId: Int,
     } else {
       // caller should have synchronized on txnMetadata already
       txnMetadata.state match {
-        case PrepareAbort | PrepareCommit =>
+        case PrepareAbort | PrepareCommit | Dead =>
```

Comment:
This is a valid point. In the `TransactionMetadata.completeTransitionTo` method, I now throw an ISE if you try to complete a transition to `Dead`. I guess we can throw the same exception here, but then the client will time out and try again.